Home page basic structure, fetchClient optional params added to make it reusable, header fixed due to request

BetterSol opened the pull request.

sviatikkondur reviewed (changes requested):
Ти чомусь видалила 16000 рядків,що це і чи воно нічого не зламає, бо бібліотеки ти ніякі тут не пушила ще?
![image](https://github.com/Hexa-Coders-aug23/product_catalog/assets/107207978/5140be34-1dbd-4cad-9e59-f10789759dc8)
 Папки для секцій - Brand New i Hot Prices не потрібні, у нас буде slider, який можна кастомізувати.
 
Також slider, видали цю папку з shared/components, ним займається Діма. Краще щоб ти скопіювала свій код з Home Page, Banner Section i Categories section, тим чим ти займаєшся, створила нову бранчу і там вже працювала, бо я не знаю, як пофіксити package-lock.json, А цей PR закрити.

BetterSol commented:
1.Це лінтер автоматично правив файл packaje-lock.json, можу просто забрати цей файл. 
2.У нас дуже багато в файлах висвітлюється через лінтер, треба буде щоб хтось одним комітом це поправив.
3.Папки Brand New i Hot Prices я ставила як заглушки, тимчасово, як буде Slider я їх видалю.
4.Папку Slider  зробила для себе, там має бути компонента Slider/Swiper/Autoplay, можу її запхати просто в HomePage.
5.Щодо файлу .api Олени - ок, я розумію, але щоб не дублювати код вирішила перевикористати наявний. @Helen-arch, тобі воно нічого не поламає?

P.S. That's what pull requests are made for ;)

sviatikkondur commented:
> 1.Це лінтер автоматично правив файл packaje-lock.json, можу просто забрати цей файл. 2.У нас дуже багато в файлах висвітлюється через лінтер, треба буде щоб хтось одним комітом це поправив. 3.Папки Brand New i Hot Prices я ставила як заглушки, тимчасово, як буде Slider я їх видалю. 4.Папку Slider зробила для себе, там має бути компонента Slider/Swiper/Autoplay, можу її запхати просто в HomePage. 5.Щодо файлу .api Олени - ок, я розумію, але щоб не дублювати код вирішила перевикористати наявний. @Helen-arch, тобі воно нічого не поламає?
> 
> P.S. That's what pull requests are made for ;)

To fix all files changing 
```git config core.autocrlf false```
```git rm --cached -r . ```
```git reset --hard```

Так, прибери package-lock-json з PR,
4. Так перенеси цей компонент в Home Page, так як він юзається тільки там

BetterSol commented:
Removed package-lock.json from commit. 
Fixed `Favourites` and `Cart` buttons in BurgerMenu.
Removed unnecessary components from HomePage and used some plugs instead.
Moved my `Slider` folder to `HomePage/components` folder (it's technically Swiper folder).
Waiting for a response from @Helen-arch, if I could make minor changes and use the `getPhones` to get the amount of phones.

sviatikkondur commented on src/api/phones.ts:
Use https://product-catalog-api-hy23.onrender.com/phones/length to get phones count, it returns only number of phones, without unnecessary information, we have the same for tablets and accessories:

phones - https://product-catalog-api-hy23.onrender.com/phones/length
tablets - https://product-catalog-api-hy23.onrender.com/tablets/length
accessories - https://product-catalog-api-hy23.onrender.com/accessories/length

Remove this file from commit also, we will use these phones only for phones on Phones page

sviatikkondur reviewed (changes requested):
Great job! Just don't use phones endpoint to get phones count!
Cool images for banner! You can continue work in this branch!

sviatikkondur reviewed (approved):
Great job!

Helen-arch reviewed (approved):
Nice job, I'm approving it!
